src/libraries/System.Net.WebSockets.Client/tests/ClientWebSocketTestBase.cs

```diff
@@ -141,14 +145,50 @@ protected static async Task<WebSocketReceiveResult> ReceiveEntireMessageAsync(We
             return null;
         }
 
+        protected Task CreateEchoServerAsync(Task clientTask, Task serverTask)
```

Comment:
This is a bit confusing now as it isn't creating anything, it's just awaiting the provided tasks.

Right now you have tests doing
```c#
(uri, serverTask) = GetServer();

await CreateEchoServerAsync(Task.Run(async () =>
{
    using ClientWebSocket cws = await GetConnectedWebSocket(uri, TimeOutMilliseconds, _output);
    // client test stuff
}, serverTask);
```

This still has the problem regarding the lifetime of the server.
> Makes managing the lifetime of the server trivial since you never directly hand it out
Currently you are returning the server uri and task, but the server itself isn't getting cleaned up

---

Have you considered making this instead be
```c#
protected Task CreateEchoServerAsync(Func<Uri, Task> clientFunc)
```
where the implementation would effectively do
```c#
await clientFunc(RemoteServerUri);
```
for remote servers, or
```c#
using var server = WebSocketHelper.GetEchoLoopbackServer(...);
await clientFunc(server.Uri);
```
for loopback ones.

The tests can then also be simplified as they don't care about the server details
```c#
await CreateEchoServerAsync(async uri =>
{
    using ClientWebSocket cws = await GetConnectedWebSocket(uri);
    // client test stuff
});
```